eth/downloader/downloader.go

```diff
@@ -1501,18 +1501,20 @@ func (d *Downloader) processHeaders(origin uint64, pivot uint64, td *big.Int) er
 						rollbackErr = err
 
 						// If some headers were inserted, track them as uncertain
-						if n > 0 && rollback == 0 {
+						if (mode == FastSync || frequency > 1) && n > 0 && rollback == 0 {
```

Comment:
Shouldn't it be `frequency==1` ? Why do we add rollbacks only when the header check is every N:th block, but not when it's every block?

Reply:
Well, if it checked every block, there's nothing to roll back. We should only roll back if we checked every 24th block for PoW, since we don't know if the previous ones were valid or not

Reply:
If we pass the pivot point, the frequency is 1. In this case, we don't need to rollback?

what about this:

```go
if frequency > 1 && n > 0 && rollback == 0 {
```